include/my_align_alloc.h

```diff
@@ -0,0 +1,58 @@
+/* Copyright (c) 2000, 2011, Oracle and/or its affiliates. 2016 IBM
+   All rights reserved.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */
+
+#ifndef MY_ALIGN_ALLOC_INCLUDED
+#define MY_ALIGN_ALLOC_INCLUDED
+
+
+#if defined (_MSC_VER)
+#include <malloc.h>
+#define ALIGNED_ALLOC(S,A) _aligned_malloc((A), (S))
+#elif defined (_ISOC11_SOURCE)
+#define ALIGNED_ALLOC(S,A)  aligned_alloc((A), (S))
```

Comment:
How does it work? According to manual: "The size parameter must be an integral multiple of alignment."

But you never verify this in your patch.